src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -2862,40 +2960,31 @@ private bool ShouldInferDeclarationExpressionValEscape(BoundExpression argument,
         /// </summary>
         private bool CheckInvocationArgMixing(
             SyntaxNode syntax,
-            in MethodInfo methodInfo,
-            BoundExpression? receiverOpt,
-            ThreeState receiverIsSubjectToCloning,
-            ImmutableArray<ParameterSymbol> parameters,
-            ImmutableArray<BoundExpression> argsOpt,
-            ImmutableArray<RefKind> argRefKindsOpt,
-            ImmutableArray<int> argsToParamsOpt,
+            in MethodInvocationInfo methodInvocationInfo,
             SafeContext localScopeDepth,
+            Symbol symbolForReporting,
```

Comment:
What would change if we continued using `methodInfo.Symbol` as previously instead of adding new `symbolForReporting` parameter?

Reply:
Error messages would use the implementation method, not the extension method.